### ISSUE TITLE
INN makefiles are not `make -j`-safe

### ISSUE THREAD

**Issue by ewxrjk** [InterNetNews/inn#206]
```
$ make -j$(nproc)
[...]
../libtool --mode=compile gcc -g -O2 -fPIE -fstack-protector-strong -I../include  -I.   -c -o ovsqlite/ovsqlite-server.lo ovsqlite/ovsqlite-server.c
libtool: compile:  gcc -g -O2 -fstack-protector-strong -I../include -I. -c ovsqlite/ovsqlite-server.c  -fPIC -DPIC -o ovsqlite/.libs/ovsqlite-server.o
ovsqlite/ovsqlite-server.c:41:14: fatal error: sql-init.h: No such file or directory
 #    include "sql-init.h"
              ^~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:82: ovsqlite/ovsqlite-server.lo] Error 1
make[1]: Leaving directory '/home/richard/src/inn/storage'
make: *** [Makefile:73: all-store-util] Error 2
make: *** Waiting for unfinished jobs....
```

The reason is that `sql-init.h` is generated at build time, so `ovsqlite/ovsqlite-server.lo` needs to explicitly depend upon `ovsqlite/sql-init.h`. This dependency is not explicitly stated in any makefile.

There _is_ a dependency from `ovsqlite/ovsqlite-server.o` on `sql-init.h` in `storage/Makefile`. This is, also, generated code, but it's generated wrongly: it should be `.lo` rather than `.o`, at least for this directory (it's fine in directories where we don't build shared libraries).

So the underlying issue is that `makedepend` is broken. (Or possibly that shared libraries are more trouble than they are worth for INN.)

**Comment by Julien-Elie** [comment]
Incidentally, there's a bug appearing in the parallel builds on Fedora, only on ppc64le (8 cores) and aarch64 (224 cores), but also for `libstorage`.  For a reason I still do not know, the build system tries to rebuild a second time `libstorage`, and removes it just before its rebuild.  During that lapse of time, another part of INN is built and fails because `libstorage` is not found.

Here is the whole build log:
[inn-2.6.4-rawhide-aarch64.log](https://github.com/InterNetNews/inn/files/7376359/inn-2.6.4-rawhide-aarch64.log)

**Comment by ewxrjk** [comment]
Oh, it's much more broken than I realized l-(

1. top-level `all` depends on `all-libraries` and `all-programs`. There are no dependencies between these so it will evaluate them concurrently.
2. `all-libraries` depends on `all-storage` which will `cd storage && make library`
3. Concurrently, `all-programs` depends on `all-innd` which will `cd innd && make all`
4. In the `innd` directory, `innd` depends on libstorage and because it hasn't been built yet, it will `cd ../storage && make library`, concurrently with #2
5. The link of libstorage, which happens at least twice without any serialization due to the above structure, starts by removing links to the real build products in the `.libs` directory, meaning that any consumers of it that get linked at the wrong moment fail to link.

The pattern is repeated throughout the various directories.

**Comment by rathann** [comment]
So, I've just got another parallel-make related failure with 2.7.0 on Fedora s390x builder:
[build.log](https://github.com/InterNetNews/inn/files/9952132/build.log) . I can reproduce another failure locally if I enable SQLite support: [bc.log](https://github.com/InterNetNews/inn/files/9952247/bc.log)

**Comment by Julien-Elie** [comment]
Does a parallel make run fine if you change in `storage/ovsqlite/ovmethod.mk`:

`ovsqlite/ovsqlite-server: $(OVSQLITEOBJECTS) $(LIBSTORAGE)`

to

`ovsqlite/ovsqlite-server: $(OVSQLITEOBJECTS) libinnstorage.$(EXTLIB)`

It is how other overview methods reference the library.

**Comment by rathann** [comment]
@Julien-Elie It looks like that last change fixed the build. I haven't seen any parallel make-related build failures for a few months now. Thank you!

**Comment by Julien-Elie** [comment]
Thanks for your feedback @rathann.
Switching the build to a non-recursive one with Automake has its own ticket (#22) so we can close this one, specific to failures in parallel builds.  In case they unfortunately occur again, feel free to open a new issue or re-open this one.
